src/client.spec.ts

```diff
@@ -17,6 +25,83 @@ describe('frontend-http-client', () => {
   beforeEach(() => mockServer.start())
   afterEach(() => mockServer.stop())
 
+  describe('sendByRouteDefinition', () => {
+    it('returns deserialized response', async () => {
+      const client = wretch(mockServer.url)
+
+      await mockServer.forPost('/users/1').thenJson(200, { data: { code: 99 } })
+
+      const requestBodySchema = z.object({
+        isActive: z.boolean(),
+      })
+
+      const responseBodySchema = z.object({
+        data: z.object({
+          code: z.number(),
+        }),
+      })
+
+      const pathSchema = z.object({
+        userId: z.number(),
+      })
+
+      const routeDefinition = buildRouteDefinition({
+        method: 'post',
+        isEmptyResponseExpected: false,
+        isNonJSONResponseExpected: false,
+        responseBodySchema,
+        requestPathParamsSchema: pathSchema,
+        requestBodySchema: requestBodySchema,
+        pathResolver: (pathParams) => `/users/${pathParams.userId}`,
+      })
+
+      const responseBody = await sendByRouteDefinition(client, routeDefinition, {
+        pathParams: {
+          userId: 1,
+        },
+        body: {
+          isActive: true,
+        },
+      })
+
+      expect(responseBody).toEqual({
+        data: {
+          code: 99,
+        },
+      })
+    })
+
+    it('returns deserialized response without body or path params', async () => {
+      const client = wretch(mockServer.url)
+
+      await mockServer.forPost('/users').thenJson(200, { data: { code: 99 } })
+
+      const responseBodySchema = z.object({
+        data: z.object({
+          code: z.number(),
+        }),
+      })
+
+      const routeDefinition = buildRouteDefinition({
+        method: 'post',
+        isEmptyResponseExpected: false,
+        isNonJSONResponseExpected: false,
+        responseBodySchema,
+        requestPathParamsSchema: undefined,
```

Comment:
this is intentionally required to be provided manually even if not set, to make sure developer did not forget to fill it by mistake, as more often than not they are going to be used.

this approach is questionable, I'm open to adjusting it if there are strong opinions about it